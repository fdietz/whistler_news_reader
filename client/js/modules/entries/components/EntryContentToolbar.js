import React, { PropTypes } from 'react';
import classNames from 'classnames';

import {
  ArrowLeftBoldSVGIcon,
  ArrowRightBoldSVGIcon,
  ArrowDownSVGIcon,
  ArrowUpSVGIcon,
  EarthSVGIcon,
  GoBackSVGIcon,
  ShareSVGIcon,
  NewspaperSVGIcon,
  EllipsisSVGIcon
} from '../../../components/SVGIcon';

import Icon from '../../../components/Icon';
import Button from '../../../components/Button';
import ButtonGroup from '../../../components/ButtonGroup';
import DropdownTrigger from '../../../components/DropdownTrigger';
import DropdownContent from '../../../components/DropdownContent';
import Dropdown from '../../../components/Dropdown';

import {
  MailLink,
  TwitterLink,
  FacebookLink,
  GooglePlusLink,
  PinterestLink,
} from './ShareButton';

const EntryContentToolbar = ({
  isMobile,
  entry,
  showSpinner,
  currentViewMode,
  hasPreviousEntry,
  hasNextEntry,
  onPreviousEntryClick,
  onNextEntryClick,
  onOpenExternalClick,
  onGoBackClick,
  onChangeViewModeClick,
}) => {
  const shareDropdownCls = classNames('btn btn-header', {
    disabled: !entry,
  });

  const spinnerCls = classNames({
    spinner: true,
    hide: !showSpinner,
  });

  const toolbarCls = classNames('toolbar', {
    'toolbar--titled': isMobile,
    'show-small-screen-only': isMobile,
    'hide-small-screen': !isMobile
  });

  const dropdownCls = classNames('dropdown', {
    west: isMobile,
    'ml1 mr1': !isMobile
  });

  return (
    <div className={toolbarCls}>
      <Button
        type="header"
        onClick={onGoBackClick}
        title="Go back"
        className="show-small-screen-only mr1">
        <GoBackSVGIcon color="light-gray" size="small" />
      </Button>

      {!isMobile &&
        <ButtonGroup className="btn-group-rounded">
          <Button
            type="header"
            onClick={onPreviousEntryClick}
            disabled={!hasPreviousEntry}
            title="Select previous story">
            <ArrowLeftBoldSVGIcon color="light-gray" size="small" />
          </Button>
          <Button
            type="header"
            onClick={onNextEntryClick}
            disabled={!hasNextEntry}
            title="Select next story">
            <ArrowRightBoldSVGIcon color="light-gray" size="small" />
          </Button>
        </ButtonGroup>
      }

      <div className="right">
        {isMobile &&
          <span className={spinnerCls}>
            <Icon name="spinner" size="small" />
          </span>
        }

        {isMobile &&
          <Button
            type="header"
            onClick={onOpenExternalClick}
            disabled={!entry}
            title="Open story in new browser tab or window">
            <EarthSVGIcon color="light-gray" size="small" />
          </Button>
        }

        <Dropdown className={dropdownCls}>
          <DropdownTrigger className="btn btn-header">
            {isMobile && <EllipsisSVGIcon color="light-gray" size="small" />}
            {!isMobile && <NewspaperSVGIcon color="light-gray" size="small" />}
            {!isMobile && <ArrowDownSVGIcon color="light-gray" size="small" className="arrow-down" />}
            {!isMobile && <ArrowUpSVGIcon color="light-gray" size="small" className="arrow-up" />}
          </DropdownTrigger>
          <DropdownContent>
            <ul className="dropdown__list">
              <li
                className="dropdown__list-item"
                onClick={onChangeViewModeClick.bind(this, 'normal')}>
                <input
                  type="radio"
                  id="list"
                  name="currrent_view_mode"
                  checked={currentViewMode === 'normal'}
                  readOnly
                  className="dropdown-field media" />
                <label
                  htmlFor="list"
                  className="dropdown-label content">Feed Article</label>
              </li>
              <li
                className="dropdown__list-item"
                onClick={onChangeViewModeClick.bind(this, 'article')}>
                <input
                  type="radio"
                  id="article"
                  name="currrent_view_mode"
                  checked={currentViewMode === 'article'}
                  readOnly
                  className="dropdown-field media" />
                <label
                  htmlFor="article"
                  className="dropdown-label content">Readability optimized (beta)</label>
              </li>
              <li
                className="dropdown__list-item"
                onClick={onChangeViewModeClick.bind(this, 'website')}>
                <input
                  type="radio"
                  id="website"
                  name="currrent_view_mode"
                  checked={currentViewMode === 'website'}
                  readOnly
                  className="dropdown-field media" />
                <label
                  htmlFor="website"
                  className="dropdown-label content">Embed Website</label>
              </li>
              {isMobile && entry &&
                <ul className="dropdown__list">
                  <li className="dropdown__list-separator" />
                  <li className="dropdown__list-item">
                    <MailLink {...entry} iconColor="light-gray" />
                  </li>
                  <li className="dropdown__list-item">
                    <TwitterLink {...entry} iconColor="light-gray" />
                  </li>
                  <li className="dropdown__list-item">
                    <FacebookLink {...entry} iconColor="light-gray" />
                  </li>
                  <li className="dropdown__list-item">
                    <GooglePlusLink {...entry} iconColor="light-gray" />
                  </li>
                  <li className="dropdown__list-item">
                    <PinterestLink {...entry} iconColor="light-gray" />
                  </li>
                </ul>
              }
            </ul>
          </DropdownContent>
        </Dropdown>

        {!isMobile &&
          <Button
            type="header"
            onClick={onOpenExternalClick}
            disabled={!entry}
            title="Open story in new browser tab or window">
            <EarthSVGIcon color="light-gray" size="small" />
          </Button>
        }
        {!isMobile &&
          <Dropdown className="ml1">
            <DropdownTrigger className={shareDropdownCls}>
              <ShareSVGIcon color="light-gray" size="small" />
              <ArrowDownSVGIcon color="light-gray" size="small" className="arrow-down" />
              <ArrowUpSVGIcon color="light-gray" size="small" className="arrow-up" />
            </DropdownTrigger>
            <DropdownContent>
            {entry &&
              <ul className="dropdown__list">
                <li className="dropdown__list-item">
                  <MailLink {...entry} />
                </li>
                <li className="dropdown__list-item">
                  <TwitterLink {...entry} />
                </li>
                <li className="dropdown__list-item">
                  <FacebookLink {...entry} />
                </li>
                <li className="dropdown__list-item">
                  <GooglePlusLink {...entry} />
                </li>
                <li className="dropdown__list-item">
                  <PinterestLink {...entry} />
                </li>
              </ul>
            }
            </DropdownContent>
          </Dropdown>
        }
      </div>
      <span className={spinnerCls}>
        <Icon name="spinner" size="small" />
      </span>
    </div>
  );
};

EntryContentToolbar.propTypes = {
  isMobile: PropTypes.bool.isRequired,
  entry: PropTypes.object,
  currentViewMode: PropTypes.string.isRequired,
  hasPreviousEntry: PropTypes.bool.isRequired,
  hasNextEntry: PropTypes.bool.isRequired,
  onPreviousEntryClick: PropTypes.func.isRequired,
  onNextEntryClick: PropTypes.func.isRequired,
  onOpenExternalClick: PropTypes.func.isRequired,
  onGoBackClick: PropTypes.func.isRequired,
  onChangeViewModeClick: PropTypes.func.isRequired,
  showSpinner: PropTypes.bool.isRequired,
};

export default EntryContentToolbar;
