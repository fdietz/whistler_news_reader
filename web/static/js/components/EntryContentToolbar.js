import React, { PropTypes } from "react";
import classNames from "classnames";

import {
  ArrowLeftBoldSVGIcon,
  ArrowRightBoldSVGIcon,
  ArrowDownSVGIcon,
  ArrowUpSVGIcon,
  EarthSVGIcon,
  GoBackSVGIcon,
  ResizeEnlargeSVGIcon,
  ShareSVGIcon
} from "../components/SVGIcon";

import Button from "../components/Button";
import ButtonGroup from "../components/ButtonGroup";
import DropdownTrigger from "../components/DropdownTrigger";
import DropdownContent from "../components/DropdownContent";
import Dropdown from "../components/Dropdown";

import {
  MailLink,
  TwitterLink,
  FacebookLink,
  GooglePlusLink,
  PinterestLink
} from "../components/ShareButton";

const EntryContentToolbar = ({
  entry,
  hasPreviousEntry,
  hasNextEntry,
  onPreviousEntryClick,
  onNextEntryClick,
  onOpenExternalClick,
  onGoBackClick
}) => {
  const shareDropdownCls = classNames("btn btn-header", {
    disabled: !entry
  });

  return (
    <div className="toolbar">
      <Button
        type="header"
        onClick={onGoBackClick}
        title="Go back"
        className="hide-medium-screen mr1">
        <GoBackSVGIcon color="light-gray" size="small"/>
      </Button>
      <ButtonGroup className="btn-group-rounded">
        <Button
          type="header"
          onClick={onPreviousEntryClick}
          disabled={!hasPreviousEntry}
          title="Select previous story">
          <ArrowLeftBoldSVGIcon color="light-gray" size="small"/>
        </Button>
        <Button
          type="header"
          onClick={onNextEntryClick}
          disabled={!hasNextEntry}
          title="Select next story">
          <ArrowRightBoldSVGIcon color="light-gray" size="small"/>
        </Button>
      </ButtonGroup>
      <ButtonGroup className="btn-group-rounded ml1">
        <Button
          type="header"
          onClick={onOpenExternalClick}
          disabled={!entry}
          title="Open story in new browser tab or window">
          <EarthSVGIcon color="light-gray" size="small"/>
        </Button>
      </ButtonGroup>
        <Dropdown className="ml1">
          <DropdownTrigger className={shareDropdownCls}>
            <ShareSVGIcon color="light-gray" size="small"/>
            <ArrowDownSVGIcon color="light-gray" size="small" className="arrow-down"/>
            <ArrowUpSVGIcon color="light-gray" size="small" className="arrow-up"/>
          </DropdownTrigger>
          <DropdownContent>
          {entry &&
            <ul className="dropdown__list">
              <li className="dropdown__list-item">
                <MailLink {...entry}/>
              </li>
              <li className="dropdown__list-item">
                <TwitterLink {...entry}/>
              </li>
              <li className="dropdown__list-item">
                <FacebookLink {...entry}/>
              </li>
              <li className="dropdown__list-item">
                <GooglePlusLink {...entry}/>
              </li>
              <li className="dropdown__list-item">
                <PinterestLink {...entry}/>
              </li>
            </ul>
          }
          </DropdownContent>
        </Dropdown>
    </div>
  );
};

EntryContentToolbar.propTypes = {
  entry: PropTypes.object,
  hasPreviousEntry: PropTypes.bool.isRequired,
  hasNextEntry: PropTypes.bool.isRequired,
  onPreviousEntryClick: PropTypes.func.isRequired,
  onNextEntryClick: PropTypes.func.isRequired,
  onOpenExternalClick: PropTypes.func.isRequired,
  onGoBackClick: PropTypes.func.isRequired
};

export default EntryContentToolbar;
