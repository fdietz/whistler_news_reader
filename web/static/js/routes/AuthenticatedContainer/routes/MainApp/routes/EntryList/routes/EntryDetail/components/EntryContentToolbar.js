import React, { PropTypes } from 'react';
import classNames from 'classnames';

import {
  ArrowLeftBoldSVGIcon,
  ArrowRightBoldSVGIcon,
  ArrowDownSVGIcon,
  ArrowUpSVGIcon,
  EarthSVGIcon,
  GoBackSVGIcon,
  TextSVGIcon,
  ShareSVGIcon,
  NewspaperSVGIcon,
  PopupSVGIcon,
} from 'components/SVGIcon';

import Icon from 'components/Icon';
import Button from 'components/Button';
import ButtonGroup from 'components/ButtonGroup';
import DropdownTrigger from 'components/DropdownTrigger';
import DropdownContent from 'components/DropdownContent';
import Dropdown from 'components/Dropdown';

import {
  MailLink,
  TwitterLink,
  FacebookLink,
  GooglePlusLink,
  PinterestLink,
} from './ShareButton';

const EntryContentToolbar = ({
  entry,
  showSpinner,
  currentViewMode,
  hasPreviousEntry,
  hasNextEntry,
  onPreviousEntryClick,
  onNextEntryClick,
  onOpenExternalClick,
  onGoBackClick,
  onChangeViewModeClick,
}) => {
  const shareDropdownCls = classNames('btn btn-header', {
    disabled: !entry,
  });

  const spinnerCls = classNames({
    spinner: true,
    hide: !showSpinner,
  });

  return (
    <div className="toolbar hide-small-screen">
      <Button
        type="header"
        onClick={onGoBackClick}
        title="Go back"
        className="show-small-screen-only mr1">
        <GoBackSVGIcon color="light-gray" size="small" />
      </Button>

      <ButtonGroup className="btn-group-rounded">
        <Button
          type="header"
          onClick={onPreviousEntryClick}
          disabled={!hasPreviousEntry}
          title="Select previous story">
          <ArrowLeftBoldSVGIcon color="light-gray" size="small" />
        </Button>
        <Button
          type="header"
          onClick={onNextEntryClick}
          disabled={!hasNextEntry}
          title="Select next story">
          <ArrowRightBoldSVGIcon color="light-gray" size="small" />
        </Button>
      </ButtonGroup>

      <Dropdown className="ml1 mr1">
        <DropdownTrigger className="btn btn-header">
          <NewspaperSVGIcon color="light-gray" size="small" />
          <ArrowDownSVGIcon color="light-gray" size="small" className="arrow-down" />
          <ArrowUpSVGIcon color="light-gray" size="small" className="arrow-up" />
        </DropdownTrigger>
        <DropdownContent>
          <ul className="dropdown__list">
            <li
              className="dropdown__list-item"
              onClick={onChangeViewModeClick.bind(this, 'normal')}>
              <input
                type="radio"
                id="list"
                name="currrent_view_mode"
                checked={currentViewMode === 'normal'}
                readOnly
                className="dropdown-field media" />
              <label
                htmlFor="list"
                className="dropdown-label content">Feed Article</label>
            </li>
            <li
              className="dropdown__list-item"
              onClick={onChangeViewModeClick.bind(this, 'article')}>
              <input
                type="radio"
                id="article"
                name="currrent_view_mode"
                checked={currentViewMode === 'article'}
                readOnly
                className="dropdown-field media" />
              <label
                htmlFor="article"
                className="dropdown-label content">Readability optimized article (experimental)</label>
            </li>
            <li
              className="dropdown__list-item"
              onClick={onChangeViewModeClick.bind(this, 'website')}>
              <input
                type="radio"
                id="website"
                name="currrent_view_mode"
                checked={currentViewMode === 'website'}
                readOnly
                className="dropdown-field media" />
              <label
                htmlFor="website"
                className="dropdown-label content">Embed Website</label>
            </li>
          </ul>
        </DropdownContent>
      </Dropdown>

      <Button
        type="header"
        onClick={onOpenExternalClick}
        disabled={!entry}
        title="Open story in new browser tab or window">
        <EarthSVGIcon color="light-gray" size="small" />
      </Button>
      <Dropdown className="ml1">
        <DropdownTrigger className={shareDropdownCls}>
          <ShareSVGIcon color="light-gray" size="small" />
          <ArrowDownSVGIcon color="light-gray" size="small" className="arrow-down" />
          <ArrowUpSVGIcon color="light-gray" size="small" className="arrow-up" />
        </DropdownTrigger>
        <DropdownContent>
        {entry &&
          <ul className="dropdown__list">
            <li className="dropdown__list-item">
              <MailLink {...entry} />
            </li>
            <li className="dropdown__list-item">
              <TwitterLink {...entry} />
            </li>
            <li className="dropdown__list-item">
              <FacebookLink {...entry} />
            </li>
            <li className="dropdown__list-item">
              <GooglePlusLink {...entry} />
            </li>
            <li className="dropdown__list-item">
              <PinterestLink {...entry} />
            </li>
          </ul>
        }
        </DropdownContent>
      </Dropdown>

      <span className={spinnerCls}>
        <Icon name="spinner" size="small" />
      </span>
    </div>
  );
};

EntryContentToolbar.propTypes = {
  entry: PropTypes.object,
  currentViewMode: PropTypes.string.isRequired,
  hasPreviousEntry: PropTypes.bool.isRequired,
  hasNextEntry: PropTypes.bool.isRequired,
  onPreviousEntryClick: PropTypes.func.isRequired,
  onNextEntryClick: PropTypes.func.isRequired,
  onOpenExternalClick: PropTypes.func.isRequired,
  onGoBackClick: PropTypes.func.isRequired,
  onChangeViewModeClick: PropTypes.func.isRequired,
};

export default EntryContentToolbar;
