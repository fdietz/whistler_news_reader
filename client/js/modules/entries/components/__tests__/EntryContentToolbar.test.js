import renderer from 'react-test-renderer';
import React from 'react';
import EntryContentToolbar from '../EntryContentToolbar';

describe('EntryContentToolbar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <EntryContentToolbar
        isMobile={true}
        showSpinner={false}
        currentViewMode="normal"
        hasPreviousEntry={true}
        hasNextEntry={true}
        onPreviousEntryClick={() => {}}
        onNextEntryClick={() => {}}
        onOpenExternalClick={() => {}}
        onGoBackClick={() => {}}
        onChangeViewModeClick={() => {}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
