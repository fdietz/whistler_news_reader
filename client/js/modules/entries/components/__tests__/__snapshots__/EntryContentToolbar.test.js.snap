exports[`EntryContentToolbar renders correctly 1`] = `
<div
  className="toolbar toolbar--titled show-small-screen-only">
  <button
    className="btn show-small-screen-only mr1 btn-header"
    onClick={[Function onGoBackClick]}
    title="Go back">
    <svg
      height={20}
      viewBox="0 0 20 20 "
      width={20}>
      <path
        d="M4.983 15l0.001-3.228 14.816 0.001v-3.553h-14.817l0.001-3.22-4.784 5z"
        fill="#98B0D2" />
    </svg>
  </button>
  <div
    className="right">
    <span
      className="spinner hide">
      <span
        className="svg-entypo-icon-spinner svg-icon-small" />
    </span>
    <button
      className="btn btn-header"
      disabled={true}
      onClick={[Function onOpenExternalClick]}
      title="Open story in new browser tab or window">
      <svg
        height={20}
        viewBox="0 0 20 20 "
        width={20}>
        <path
          d="M10 0.399c-5.294 0-9.6 4.308-9.6 9.601 0 5.294 4.306 9.601 9.6 9.601s9.6-4.307 9.6-9.601c0-5.293-4.306-9.601-9.6-9.601zM18.188 10c0 1.873-0.636 3.601-1.696 4.981-0.3-0.235-0.618-0.868-0.318-1.524 0.302-0.66 0.381-2.188 0.312-2.782-0.066-0.594-0.375-2.026-1.215-2.040-0.837-0.013-1.412-0.289-1.91-1.283-1.033-2.067 1.939-2.465 0.906-3.61-0.289-0.321-1.783 1.324-2.002-0.868-0.015-0.157 0.135-0.392 0.335-0.635 3.244 1.090 5.588 4.156 5.588 7.761zM8.875 1.894c-0.195 0.38-0.712 0.536-1.027 0.823-0.683 0.618-0.977 0.533-1.346 1.127s-1.566 1.449-1.566 1.879c0 0.429 0.604 0.937 0.906 0.838 0.302-0.1 1.098-0.094 1.566 0.070 0.469 0.166 3.915 0.331 2.816 3.244-0.348 0.926-1.872 0.77-2.278 2.304-0.061 0.225-0.271 1.186-0.286 1.5-0.024 0.486 0.344 2.318-0.125 2.318-0.471 0-1.737-1.639-1.737-1.936s-0.329-1.339-0.329-2.23-1.517-0.877-1.517-2.062c0-1.070 0.823-1.602 0.638-2.113-0.182-0.51-1.627-0.528-2.229-0.59 1.052-2.732 3.529-4.76 6.514-5.172zM7.425 17.77c0.492-0.259 0.542-0.595 0.988-0.613 0.51-0.022 0.925-0.199 1.499-0.325 0.511-0.111 1.423-0.629 2.228-0.695 0.677-0.056 2.015 0.035 2.375 0.689-1.296 0.86-2.848 1.362-4.515 1.362-0.9 0-1.765-0.148-2.575-0.418z"
          fill="#98B0D2" />
      </svg>
    </button>
    <div
      className="dropdown dropdown west"
      style={undefined}>
      <a
        className="dropdown__trigger btn btn-header"
        onClick={[Function bound toggleActive]}
        style={undefined}>
        <svg
          height={20}
          viewBox="0 0 20 20 "
          width={20}>
          <path
            d="M3 7.8c-1.215 0-2.2 0.985-2.2 2.2s0.985 2.2 2.2 2.2 2.2-0.985 2.2-2.2c0-1.215-0.985-2.2-2.2-2.2zM10 7.8c-1.215 0-2.2 0.985-2.2 2.2s0.985 2.2 2.2 2.2 2.199-0.985 2.199-2.2c0-1.215-0.984-2.2-2.199-2.2zM17 7.8c-1.215 0-2.2 0.985-2.2 2.2s0.985 2.2 2.2 2.2 2.2-0.985 2.2-2.2c0-1.215-0.985-2.2-2.2-2.2z"
            fill="#98B0D2" />
        </svg>
      </a>
      <div
        className="dropdown__content hidden"
        onClick={[Function bound handleClickInsideDropdownContent]}
        style={undefined}>
        <ul
          className="dropdown__list">
          <li
            className="dropdown__list-item"
            onClick={[Function bound onChangeViewModeClick]}>
            <input
              checked={true}
              className="dropdown-field media"
              id="list"
              name="currrent_view_mode"
              readOnly={true}
              type="radio" />
            <label
              className="dropdown-label content"
              htmlFor="list">
              Feed Article
            </label>
          </li>
          <li
            className="dropdown__list-item"
            onClick={[Function bound onChangeViewModeClick]}>
            <input
              checked={false}
              className="dropdown-field media"
              id="article"
              name="currrent_view_mode"
              readOnly={true}
              type="radio" />
            <label
              className="dropdown-label content"
              htmlFor="article">
              Readability optimized (beta)
            </label>
          </li>
          <li
            className="dropdown__list-item"
            onClick={[Function bound onChangeViewModeClick]}>
            <input
              checked={false}
              className="dropdown-field media"
              id="website"
              name="currrent_view_mode"
              readOnly={true}
              type="radio" />
            <label
              className="dropdown-label content"
              htmlFor="website">
              Embed Website
            </label>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <span
    className="spinner hide">
    <span
      className="svg-entypo-icon-spinner svg-icon-small" />
  </span>
</div>
`;
